### PR TITLE
DOC: add documentation for Meson arguments

### PR DESCRIPTION
**Pull request by peter-urban**
This is a draft for the (how-to) meson-args documentation. It includes an example for using meson install_tags using meson-python's install options (PR https://github.com/mesonbuild/meson-python/pull/288 and issue https://github.com/mesonbuild/meson-python/issues/68)

I could not yet answer all todos
(Mention the order in which Meson arguments are added, and how it affect Meson)
and
(Explain how you can add extra Meson arguments via the build config settings)

Feel free to comment on what you like/dislike or directly copy/modify it to you liking

### CODE REVIEW

File: docs/how-to-guides/meson-args.rst
Comment (rgommers) [inline docs/how-to-guides/meson-args.rst:91]: I think this is (untested):
```
# with pypa/build
python -m build -Cinstall="--tags=runtime,python-runtime"

# with pip
python -m pip --config-settings=install="--tags=runtime,python-runtime"
```

Comment (peter-urban) [reply docs/how-to-guides/meson-args.rst:91]: Thanks, is was close enough  :-) 
I jest tested the pip command and it is:
python -m pip install --config-settings=install-args="--tags=devel,runtime,python-runtime" .

the python -m build commend fails for me for some reason (ModuleNotFoundError: No module named 'mesonpy._dylib'), I'll investigate this soon and than update this

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:39]: Please wrap key names in ``. Also, I think the brackets in `tool.meson-python.args` can be omitted.

---

> Currently all four option categories [...] must be present

Uh, this is a major oversight, that should definitely not be the case :sweat_smile:

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:79]: Same as the other `code-block` comment.

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:58]: We should probably use italic instead of quotes for *static*.

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:85]: meson-python should be wrapped in ``.

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:90]: `runtime`, `devel`, `meson install --tags=tag1,tag2,...`, and `install_tag` should be wrapped in `` instead.

> Ffor more information

Typo.

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:92]: `runtime` and `python-runtime` should be wrapped in `` instead.

> c++ headers

The C in C++ should be capitalized.

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:55]: Please split the commands into separate code-blocks, and add a `$` at the start (it gets ignored when selecting it).

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:110]: Same as the other `code-block` comment.

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:67]: We should probably use single-quotes here, to be consistent.

**Review by FFY00 — COMMENTED** [review]
I have some comments, these are all kind-of nitpicks. If they are too much for you, I can definitely fix them before merging.

Regarding the pypa/build vs pip commands. We could actually use https://sphinx-design.readthedocs.io/en/latest/tabs.html instead, I think it would look better. They should probably be synchronized tabs too.

**Comment by FFY00** [comment]
Regardless, this is very much appreciated @peter-urban!

**Comment by peter-urban** [comment]
Hi @FFY00, your comments are appreciated! 
I am planning to use sphinx in my own project and I am happy to learn some things about styling.

I'll check through your comments in the coming days and apply them one after the other. 
(I don't think there is a rush here right?)

File: docs/how-to-guides/meson-args.rst
Comment (peter-urban) [reply docs/how-to-guides/meson-args.rst:39]: thought so actually, but it didn't bother me enough to fix it :-D
... I created an issue, (https://github.com/mesonbuild/meson-python/issues/293)
If have time in the future I will see if I can make a PR forthis

Comment (peter-urban) [reply docs/how-to-guides/meson-args.rst:39]: > Also, I think the brackets in `tool.meson-python.args` can be omitted.

I get: ``ERROR Failed to parse ./pyproject.toml: Expected '=' after a key in a key/value pair (at line 91, column 23) `` if I remove the brackets arrount ``tool.meson-python.args``?

Comment (peter-urban) [reply docs/how-to-guides/meson-args.rst:39]: > Please wrap key names in ``

You mean like this? I haven't seen this before (admittedly I haven't seem many pyproject.toml files yet). Is there a reason to use the " wrapping instead of just writing the key?

**Comment by peter-urban** [comment]
Ok, I had suprisingly much time tonight :-D
It's ready for another revision round.

**Comment by peter-urban** [comment]
@FFY00 ... sorry I didn't want to be pushy with the 're-review' request. I mistook the double arrow symbol next to your name as "update"

**Comment by peter-urban** [comment]
> Regarding the pypa/build vs pip commands. We could actually use https://sphinx-design.readthedocs.io/en/latest/tabs.html instead, I think it would look better. They should probably be synchronized tabs too.

I rewrote the examples using the sphinx-design tabs command

**Comment by FFY00** [comment]
> @FFY00 ... sorry I didn't want to be pushy with the 're-review' request. I mistook the double arrow symbol next to your name as "update"

No worries at all! FYI pinging me often if something is blocked on me is actually helpful, so not bothered at all by pings.

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:42]: I think this bit should not be in the tab. IMO we should show how to configure it permanently in `pyproject.toml`, and then how to add more options (temporarily) at build time, with the user's preferred tooling, pypa/build and pip, which is what should be in a tab because they do the same thing.

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:81]: Same. I'd say something like "permanently" and then "overwriting at build time".

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:137]: nitpick: can you wrap this in 80 character lines?

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:67]: Can we have an "Example" section and then "1) Set the ..." and "2) Use Meson ..." subsections? IMO that'd look better.

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:98]: Please capitalize "Meson", as that's the right capitalization for the project :blush:

**Review by FFY00 — COMMENTED** [review]
@peter-urban thank you so much! This is great!

I just have some final comments, and then we can go ahead and merge this.

Also, if you could remove the points you tackled in the PR from the `todo` list at the top of the page, that'd be appreciated.

File: docs/how-to-guides/meson-args.rst
Comment (peter-urban) [reply docs/how-to-guides/meson-args.rst:98]: > No worries at all! FYI pinging me often if something is blocked on me is actually helpful, so not bothered at all by pings.

Good to know :-) And thanks for the comments. I'll implement them once I find some time (in the coming days).

**Comment by peter-urban** [comment]
I adapted your suggestions. Let me know what you think.

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:27]:
```suggestion
.. code-block:: toml
```

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:71]:
```suggestion
.. code-block:: toml
```

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:108]:
```suggestion
.. code-block:: toml
```

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:156]:
```suggestion
.. code-block:: toml
```

File: docs/how-to-guides/meson-args.rst
Comment (FFY00) [inline docs/how-to-guides/meson-args.rst:160]: To stay consistent.

```suggestion
		setup = ['-Doptimization=3;]
```

**Review by FFY00 — APPROVED** [review]
Thanks @peter-urban, looks great!

There are just some minor fixups, but I'll apply those via Github's UI and mege :blush:

**Pull request merged**
